packages/runner/src/runner.ts

```diff
@@ -475,4 +474,346 @@ export class Runner {
+  ): TransactionSkeletonType {
+    const getInputKey = (input: Cell) =>
+      `${input.out_point!.tx_hash}_${input.out_point!.index}`;
+    let previousInputs = Set<string>();
+    let inputCkbCapacitySum = BigInt(0);
+    let outputCkbCapacitySumForSudtCustodianCells = BigInt(0);
+    for (let [sudtScriptHash, targetSudtAmount] of sudtWithdrawalAssets) {
+      let inputSudtAmountSum = BigInt(0);
+      for (const cell of validCustodianCells) {
+        if (
+          cell.cell_output.type ||
+          cell.cell_output.type!.args === sudtScriptHash
+        ) {
+          const key = getInputKey(cell);
+          if (previousInputs.has(key)) {
+            continue;
+          }
+          previousInputs.add(key);
+          const inputCkbCapacity = BigInt(cell.cell_output.capacity);
+          const inputSudtAmount = utils.readBigUInt128LE(cell.data);
+          inputCkbCapacitySum += inputCkbCapacity;
+          inputSudtAmountSum += inputSudtAmount;
+          txSkeleton = txSkeleton.update("inputs", (inputs) => {
+            return inputs.push(cell);
+          });
+          if (inputSudtAmountSum >= targetSudtAmount) {
+            break;
+          }
+        }
+      }
+      if (inputSudtAmountSum < targetSudtAmount) {
+        throw new Error("Insufficient sudt amount in valid custodian cells");
+      }
+      // build sudt change custodian cell
+      const custodianLock: Script = {
+        code_hash: this._deploymentConfig().custodian_lock.code_hash,
+        hash_type: this._deploymentConfig().custodian_lock.hash_type,
+        args: DEFAULT_CUSTODIAN_LOCK_ARGS,
+      };
+      const custodianType: Script = {
+        code_hash: this._deploymentConfig().sudt_type.code_hash,
+        hash_type: this._deploymentConfig().sudt_type.hash_type,
+        args: sudtScriptHash,
+      };
+      let sudtChangeCustodian: Cell = {
+        cell_output: {
+          lock: custodianLock,
+          type: custodianType,
+          capacity: "0x0",
+        },
+        data: utils.toBigUInt128LE(
+          BigInt(inputSudtAmountSum) - BigInt(targetSudtAmount)
+        ),
+      };
+      const minimalCapacity = minimalCellCapacity(sudtChangeCustodian);
+      sudtChangeCustodian.cell_output.capacity =
+        "0x" + minimalCapacity.toString(16);
+      outputCkbCapacitySumForSudtCustodianCells += minimalCapacity;
+      txSkeleton = txSkeleton.update("outputs", (outputs) => {
+        return outputs.push(sudtChangeCustodian);
+      });
+    }
+    if (
+      BigInt(inputCkbCapacitySum) ===
+      BigInt(ckbWithdrawalCapacity) +
+        BigInt(outputCkbCapacitySumForSudtCustodianCells)
+    ) {
+      // so lucky with it, just return
+      return txSkeleton;
+    }
+    // build ckb change custodian cell
+    const custodianLock: Script = {
+      code_hash: this._deploymentConfig().custodian_lock.code_hash,
+      hash_type: this._deploymentConfig().custodian_lock.hash_type,
+      args: DEFAULT_CUSTODIAN_LOCK_ARGS,
```

Comment:
If this is a change cell, can we just leverage lock args from the original custodian cell?

Reply:
After future discussion, we agreed with the default value approach.